packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts

```diff
@@ -213,6 +233,8 @@ export class Schedule extends Resource implements ISchedule {
     return this.metricAll('InvocationsSentToDeadLetterCount_Truncated_MessageSizeExceeded', props);
   }
 
+  private static readonly ISO8601_REGEX = /^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])\.[0-9]{3}(Z)?$/;
```

Comment:
```suggestion
  private static readonly ISO8601_REGEX = /^([0-2]\d{3})-(0[0-9]|1[0-2])-([0-2]\d|3[01])T([01]\d|2[0-4]):([0-5]\d):([0-6]\d)((\.\d{3})?)Z$/;
```
A [previous issue](https://github.com/aws/aws-cdk/issues/28124) was caused by [TransitionDate](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-transition.html#cfn-s3-bucket-transition-transitiondate) being formatted without the final `Z` so a different regex may be better in this case.

Reply:
You are absolutely right.
I made some modifications from your suggestion because CFn throws the following error if it does not include milliseconds.
`StartDate needs to follow the format yyyy-MM-ddTHH:mm:ss.SSSZ`